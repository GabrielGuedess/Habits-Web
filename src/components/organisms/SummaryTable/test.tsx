import { render, screen } from '@testing-library/react';

import { SummaryTable } from '.';

jest.mock('components/atoms/HabitDay', () => ({
  __esModule: true,
  HabitDay: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="HabitDay Mock">{children}</div>;
  },
}));

jest.mock('components/atoms/WeekDay', () => ({
  __esModule: true,
  WeekDay: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="WeekDay Mock">{children}</div>;
  },
}));

describe('<SummaryTable />', () => {
  it('should render the SummaryTable', () => {
    const { container } = render(<SummaryTable />);

    expect(screen.getAllByTestId('HabitDay Mock')).toHaveLength(18 * 7);
    expect(screen.getAllByTestId('WeekDay Mock')).toHaveLength(7);

    expect(container.firstChild).toMatchSnapshot();
  });
});
