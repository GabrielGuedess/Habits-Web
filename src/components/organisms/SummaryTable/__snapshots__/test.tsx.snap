// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryTable /> should render the SummaryTable 1`] = `
<section
  class="flex w-full"
>
  <aside
    class="grid grid-flow-row grid-rows-7 gap-3"
  >
    <div
      data-testid="WeekDay Mock"
    />
    <div
      data-testid="WeekDay Mock"
    />
    <div
      data-testid="WeekDay Mock"
    />
    <div
      data-testid="WeekDay Mock"
    />
    <div
      data-testid="WeekDay Mock"
    />
    <div
      data-testid="WeekDay Mock"
    />
    <div
      data-testid="WeekDay Mock"
    />
  </aside>
  <main
    class="grid grid-flow-col grid-rows-7 gap-3"
  >
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
    <div
      data-testid="HabitDay Mock"
    />
  </main>
</section>
`;
